rimage/lazy_encoded.go

```diff
@@ -60,17 +64,29 @@ func (lei *LazyEncodedImage) RawData() []byte {
 
 // ColorModel returns the Image's color model.
 func (lei *LazyEncodedImage) ColorModel() color.Model {
-	lei.decode()
-	return lei.decodedImage.ColorModel()
+	if lei.colorModel == nil {
+		reader := bytes.NewReader(lei.imgBytes)
+		header, _, err := image.DecodeConfig(reader)
+		if err != nil {
+			panic(fmt.Sprintf("error extracting colorModel from lazyEncodedImage: %v", err))
+		}
+		lei.colorModel = header.ColorModel
+	}
+	return lei.colorModel
```

Comment:
much like decode() above, create a helper function called decodeConfig() which follows the same kind of style. Add a new field in the struct called `decodeConfigOnce sync.Once` and create the Do command which will call DecodeConfig. If the helper function panics, store the message in decodeErr.

Essentially, follow the same kind of pattern that `At()` does for decoding the entire image.